flexible_event_config.md

```diff
@@ -47,7 +50,154 @@ In general, the approach here is to more flexibly encode the output of the API f
 
 ### API changes
 
-In addition to the parameters that were added in Phase 1, we will add one additional optional parameter to the JSON in `Attribution-Reporting-Register-Source`: `trigger_specs`
+In addition to the parameters that were added in Phase 1 to the JSON in `Attribution-Reporting-Register-Source`, sources can now be configured to allow non-default `trigger_data` (values and/or
+cardinality):
+
+```jsonc
+{
+  ...
+
+  // Specifies how the 64-bit unsigned trigger_data from the trigger is matched
+  // against the source's trigger_data, which is 32-bit. Defaults to "modulus".
+  //
+  // If "exact", the trigger_data must exactly match a value contained in the
+  // source's trigger_data; if there is no such match, no event-level
+  // attribution takes place.
+  //
+  // If "modulus", the source's trigger_data must form a contiguous sequence of
+  // integers starting at 0. The trigger's trigger_data is taken modulus the
+  // cardinality of this sequence and then matched against the trigger data.
+  // See below for an example. It is an error to use "modulus" if the trigger
+  // data does not form such a sequence.
+  "trigger_data_matching": <one of "exact" or "modulus">,
+
+  // Size must be in the range [0, 32], inclusive.
+  // If omitted, defaults to [0, 1, 2, 3, 4, 5, 6, 7] for navigation sources and
+  // [0, 1] for event sources.
+  "trigger_data": [<32-bit unsigned integer>, ...],
+
+  // See description in
+  // https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#optional-varying-frequency-and-number-of-reports
+  "max_event_level_reports": <int>,
+
+  // See description in
+  // https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#optional-varying-frequency-and-number-of-reports
+  "event_report_windows": {
+    "start_time": <int>,
+    "end_times": [<int>, ...]
+  }
+}
+```
+
+This configuration fully specifies the output space of the event-level reports, per source registration.
+
+### Trigger-data modulus matching example
+
+Given a source with the following registration:
+
+```jsonc
+{
+  "trigger_data_matching": "modulus",
+
+  "trigger_data": [0, 1, 2, 3, 4, 5],
+
+  ...
+}
+```
+
+The trigger-data cardinality is 6, so all triggers' `trigger_data` will be taken
+modulus 6 before determining the matching source:
+
+- `{"trigger_data": "0"}` will match the source because `0 % 6 = 0`
+- `{"trigger_data": "1"}` will match the source because `1 % 6 = 1`
+- `{"trigger_data": "2"}` will the source because `2 % 6 = 2`
+- `{"trigger_data": "6"}` will match Spec A because `6 % 6 = 0`
```

Comment:
Should "Spec A" and "Spec C" be 'the source,' or are these references to other, missing, example registered sources?

Reply:
Good find. Looks like something was lost in the translation here from the future idea of multiple trigger specs. I'll tidy this up.